### PR TITLE
re #1182 replacing direct string compare with case insensitive compare

### PR DESCRIPTION
**Pull request by adamrankin**
Automated tests pass, but could someone take a look?

### CODE REVIEW

File: src/PlusUsSimulator/PlusSpatialModel.cxx [hunk 1]
```diff
@@ -518,14 +521,13 @@ PlusStatus PlusSpatialModel::UpdateModelFile()
 }
 
 //-----------------------------------------------------------------------------
-void PlusSpatialModel::SetModelFile(const char* modelFile)
+void PlusSpatialModel::SetModelFile(const std::string& modelFile)
 {
-  std::string oldModelFile = this->ModelFile;
-  this->ModelFile = modelFile ? modelFile : "";
-  if (this->ModelFile.compare(oldModelFile) != 0)
+  if (!PlusCommon::IsEqualInsensitive(modelFile, this->ModelFile))
```
Comment (lassoan) [inline src/PlusUsSimulator/PlusSpatialModel.cxx]: file names are case-sensitive on non-Windows systems, I would keep case-sensitive compare

File: src/PlusCommon/PlusCommon.h [hunk 2]
```diff
@@ -47,13 +47,13 @@ enum PlusImagingMode
 
 /* Define case insensitive string compare for Windows. */
 #if defined( _WIN32 ) && !defined(__CYGWIN__)
-#  if defined(__BORLANDC__)
-#    define STRCASECMP stricmp
-#  else
-#    define STRCASECMP _stricmp
-#  endif
+  #if defined(__BORLANDC__)
+    #define STRCASECMP stricmp
```
Comment (lassoan) [inline src/PlusCommon/PlusCommon.h]: Keep only one way of case-insensitive string compare. I'm OK with the new method but then remove the STRCASECMP macro.

File: src/PlusCommon/PlusCommon.cxx [hunk 3]
```diff
@@ -612,6 +622,41 @@ bool vtkPlusCommonExport PlusCommon::IsEqualInsensitive(std::string const& a, st
 }
 
 //----------------------------------------------------------------------------
+bool vtkPlusCommonExport PlusCommon::IsEqualInsensitive(std::wstring const& a, std::wstring const& b)
```
Comment (lassoan) [inline src/PlusCommon/PlusCommon.cxx]: Have you tried if you see a difference in loading time with the original version and after this pull request?
Try to load a header with at least a few ten thousand transforms.

File: src/PlusCommon/PlusCommon.h (hunk 2 above)
Comment (adamrankin) [inline src/PlusCommon/PlusCommon.h]: There was no actual change here, just my auto-formatting changing the indent.

Comment (lassoan) [inline src/PlusCommon/PlusCommon.h]: I know. The issue is that there are multiple case-insensitive string comparison implementation in PlusCommon for us to use, maintain, document. We should keep only one.

Comment (adamrankin) [inline src/PlusCommon/PlusCommon.h]: Right, gotcha. I view the STRCASECMP for usage with C style strings and new methods for C++ strings. I suppose compilers will automatically construct C++ from C. Still replace?

Comment (lassoan) [inline src/PlusCommon/PlusCommon.h]: As we move towards std::string, we should probably retire STRCASECMP.
It is mostly used for XML attribute value checks, such as 

`if (toolDataElement->GetAttribute("Type") != NULL && STRCASECMP(toolDataElement->GetAttribute("Type"), "Tool") != 0)`

If we add a helper method for that then the code will be more compact and we can control at one place how this comparison is implemented.

For the rest of the STRCASECMP we need to check, but I guess we should be able to replace them with the new method.

File: src/PlusCommon/PlusCommon.cxx (hunk 3 above)
Comment (adamrankin) [inline src/PlusCommon/PlusCommon.cxx]: 0.296s vs. 0.261s for 23399 frames.

File: src/PlusCommon/PlusCommon.h (hunk 2 above)
Comment (adamrankin) [inline src/PlusCommon/PlusCommon.h]: Implementing this cleanup, do not accept PR until next changes.

**Pull request merged**